quil-rs/src/program/mod.rs

```diff
@@ -217,36 +238,64 @@ impl Program {
         Ok(expanded_program)
     }
 
+    pub fn is_protoquil(&self) -> Result<()> {
```

Comment:
Is a `Result<()>` an appropriate return type for this kind of function? The `Result` allows us to attach valuable context to the error but makes a `bool` success variant redundant since it would never be `false`.

Reply:
Yes, this is how I'd do it.

(that being said, see my other note about whether we should have this function at all)